infrastructures/infrastructure-azure/src/test/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructureTest.java

```diff
@@ -0,0 +1,353 @@
+
+        verify(connectorIaasController, times(2)).executeScriptWithCredentials(anyString(),
+                                                                               anyString(),
+                                                                               anyList(),
+                                                                               anyString(),
+                                                                               anyString());
+    }
+
+    @Test
+    public void testAcquireAllNodes() {
+        testAcquireNode();
+    }
+
+    @Test
+    public void testRemoveNode() throws ProActiveException, RMException {
+        when(nodeSource.getName()).thenReturn("Node source Name");
+        azureInfrastructure.nodeSource = nodeSource;
+
+        azureInfrastructure.configure("clientId",
+                                      "secret",
+                                      "domain",
+                                      "subscriptionId",
+                                      "authenticationEndpoint",
+                                      "managementEndpoint",
+                                      "resourceManagerEndpoint",
+                                      "graphEndpoint",
+                                      "test.activeeon.com",
+                                      "http://localhost:8088/connector-iaas",
+                                      "image",
+                                      "Standard_D1_v2",
+                                      "vmUsername",
+                                      "vmPassword",
+                                      "vmPublicKey",
+                                      "resourceGroup",
+                                      "region",
+                                      "2",
+                                      "3",
+                                      "wget -nv test.activeeon.com/rest/node.jar",
+                                      "192.168.1.0/24",
+                                      true,
+                                      "-Dnew=value");
+
+        azureInfrastructure.connectorIaasController = connectorIaasController;
+
+        when(node.getProperty(AzureInfrastructure.INSTANCE_ID_NODE_PROPERTY)).thenReturn("123");
+
+        when(node.getNodeInformation()).thenReturn(nodeInformation);
+
+        when(node.getProActiveRuntime()).thenReturn(proActiveRuntime);
+
+        when(nodeInformation.getName()).thenReturn("nodename");
+
+        azureInfrastructure.nodesPerInstances.put("123", Sets.newHashSet("nodename"));
+
+        azureInfrastructure.removeNode(node);
+
+        verify(proActiveRuntime).killNode("nodename");
+
+        verify(connectorIaasController).terminateInstance("node_source_name", "123");
+
+        assertThat(azureInfrastructure.nodesPerInstances.isEmpty(), is(true));
+
+    }
+
+    @Test
+    public void testNotifyAcquiredNode() throws ProActiveException, RMException {
```

Comment:
renamed to `testThatNotifyAcquiredNodeMethodFillsTheNodesMapCorrectly` in 06b250351a3cac447f5c0b0272c314c5bd90a0cb